wicket-native-websocket/wicket-native-websocket-core/src/main/java/org/apache/wicket/protocol/ws/api/message/TextMessage.java

```diff
@@ -16,19 +16,46 @@
  */
 package org.apache.wicket.protocol.ws.api.message;
 
+import org.apache.wicket.Application;
+import org.apache.wicket.mock.MockApplication;
+import org.apache.wicket.protocol.ws.api.registry.IKey;
 import org.apache.wicket.util.lang.Args;
 
 /**
  * A {@link IWebSocketMessage message} with text data
  *
  * @since 6.0
  */
-public class TextMessage implements IWebSocketMessage
+public class TextMessage extends AbstractClientMessage
 {
 	private final CharSequence text;
 
-	public TextMessage(final CharSequence text)
+	/**
+	 * Not used by Wicket since 8.5.0!
+	 *
+	 * @param text
+	 * 		the message sent from the client
+	 */
+	@Deprecated
+	public TextMessage(CharSequence text)
 	{
+		this(new MockApplication(), "", new IKey() {}, text);
```

Comment:
I guess you can't use Application.get() because can't be sure there is thread local Application attached to thread? Would it make sense to do this with some "static"  

```
ty {
 return Application.get()  
} catch() {
  return new MockApplication():
}
```

Reply:
Wicket uses the new constructors at `org.apache.wicket.protocol.ws.api.AbstractWebSocketProcessor#onMessage()` and here there is no Application ThreadLocal for sure. But let's assume that there are Application and Session thread locals - the IKey would be still unknown, and this makes all attempts useless. It would only work if the developer overrides the getter methods of Text/BinaryMessage, but then it would be better to migrate to the new constructors.

Reply:
ok